portal-ui/src/screens/Console/Policies/PolicyDetails.tsx

```diff
@@ -0,0 +1,349 @@
+        marginBottom: 67,
+      },
+      "& a": {
+        textDecoration: "none",
+      },
+      "& h3": {
+        color: "#FFFFFF",
+        marginBottom: "30px",
+        fontWeight: "bold",
+      },
+      "& h6": {
+        color: "#FFFFFF !important",
+      },
+    },
+    licenseInfo: { color: "#FFFFFF", position: "relative" },
+    licenseInfoTitle: {
+      textTransform: "none",
+      color: "#BFBFBF",
+      fontSize: 11,
+    },
+    licenseInfoValue: {
+      textTransform: "none",
+      fontSize: 14,
+      fontWeight: "bold",
+    },
+    verifiedIcon: {
+      width: 96,
+      position: "absolute",
+      right: 0,
+      bottom: 29,
+    },
+    breadcrumLink: {
+      textDecoration: "none",
+      color: "black",
+    },
+    ...modalBasic,
+    ...containerForHeader(theme.spacing(4)),
+  });
+
+const PolicyDetails = ({
+  classes,
+  match,
+  closeModalAndRefresh,
+  setErrorSnackMessage,
+}: IPolicyDetailsProps) => {
+  const [selectedTab, setSelectedTab] = useState<number>(0);
+  const [policy, setPolicy] = useState<Policy | null>(null);
+  const [userList, setUserList] = useState<User[]>([]);
+  const [addLoading, setAddLoading] = useState<boolean>(false);
+  const [policyName, setPolicyName] = useState<string>(
+    match.params["policyName"]
+  );
+  const [policyDefinition, setPolicyDefinition] = useState<string>("");
+  const [loadingPolicy, setLoadingPolicy] = useState<boolean>(true);
+  const [loadingUsers, setLoadingUsers] = useState<boolean>(true);
+
+  const loadUsersForPolicy = () => {
+    if (loadingUsers) {
+      api
+        .invoke("GET", `/api/v1/policies/${policyName}/users`)
+        .then((result: any) => {
+          setUserList(result);
+          setLoadingUsers(false);
+        })
+        .catch((err) => {
+          console.log("Error in loading users");
+        });
+    }
+  };
+  const loadPolicyDetails = () => {
+    if (loadingPolicy) {
+      api
+        .invoke("GET", `/api/v1/policy?name=${policyName}`)
+        .then((result: any) => {
+          setPolicy(result);
+          setLoadingPolicy(false);
+        })
+        .catch((err) => {
+          console.log("Error in loading policy");
+        });
+    }
+  };
+
+  const addRecord = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (addLoading) {
+      return;
+    }
+    setAddLoading(true);
+    api
+      .invoke("POST", "/api/v1/policies", {
+        name: policyName,
+        policy: policyDefinition,
+      })
+      .then((res) => {
+        setAddLoading(false);
+        closeModalAndRefresh(true);
+      })
+      .catch((err) => {
+        setAddLoading(false);
+        setModalErrorSnackMessage(err);
+      });
+  };
+
+  useEffect(() => {
+    loadUsersForPolicy();
+    loadPolicyDetails();
+    if (policy) {
+      setPolicyName(policy.name);
+      setPolicyDefinition(
+        policy ? JSON.stringify(JSON.parse(policy.policy), null, 4) : ""
+      );
+    }
+  }, [policy]);
```

Comment:
Please split this effect in two parts because there is still a couple of re-renders present:
```suggestion
useEffect(() => {
    if(loadingUsers) {
api
        .invoke("GET", `/api/v1/policies/${policyName}/users`)
        .then((result: any) => {
          setUserList(result);
          setLoadingUsers(false);
        })
        .catch((err) => {
          console.log("Error in loading users");
        });
    }
    
    if(loadingPolicy) {
      api
        .invoke("GET", `/api/v1/policy?name=${policyName}`)
        .then((result: any) => {
          setPolicy(result);
          setLoadingPolicy(false);
        })
        .catch((err) => {
          console.log("Error in loading policy");
        });
    }
}, [loadingUsers, loadingPolicy, policyName]);

  useEffect(() => {
    if (policy) {
      setPolicyName(policy.name);
      setPolicyDefinition(
        policy ? JSON.stringify(JSON.parse(policy.policy), null, 4) : ""
      );
    }
  }, [policy]);
```